Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -531,6 +634,17 @@ struct VersionAssignmentSet<C: PackageContainer>: Equatable, Sequence {
             }
             return false
 
+        case .revision(let identifier):
+            // If we already have a revision constraint, it should be same as the one
+            // we're trying to set.
+            if case .revision(let existingRevision) = constraints[container.identifier] {
+                return existingRevision == identifier
+            }
+            // Otherwise, it is always valid to set a revision binding.  Note
+            // that there are rules that prevents versioned constaints from
+            // having revision constaints, but that is handled by the resolver.
```

Comment:
I'm not sure I understand this -- why shouldn't this return false if there is an existing version constraint; this seems to violate the API contract?